src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java

```diff
@@ -164,17 +166,26 @@ public void onDestroy() {
 
         PreferencesUtils.unregister(this, sharedPreferenceChangeListener);
 
+
         try {
             voiceExecutor.shutdown();
         } finally {
             voiceExecutor = null;
         }
 
+        if (PreferencesUtils.shouldInstantExportAfterWorkout(getApplicationContext())) {
+            SharedPreferences sharedPreferences = PreferencesUtils.getSharedPreferences(this);
+            TrackFileFormat trackFileFormat = PreferencesUtils.getExportTrackFileFormat(this);
+            InstantPostWorkoutExport instantPostWorkoutExport = new InstantPostWorkoutExport(getApplicationContext(), contentProviderUtils, sharedPreferences, trackFileFormat);
+            instantPostWorkoutExport.exportLastTrackToExternalStorage(getString(R.string.settings_single_export_directory_key));
```

Comment:
use PreferencesUtils

Reply:
what do you mean by this?

Reply:
Do not pass the sharedPreferences to InstantPostWorkoutExport to retrieve infos.
Just put helper functions into PreferenceUtils that get you the data you need.